src/com/sap/piper/WhitesourceConfigurationHelper.groovy

```diff
@@ -0,0 +1,143 @@
+package com.sap.piper
+
+import com.cloudbees.groovy.cps.NonCPS
+
+class WhitesourceConfigurationHelper implements Serializable {
+
+    static def extendUAConfigurationFile(script, utils, config, path) {
+        def mapping = []
+        def parsingClosure = { fileReadPath -> return script.readProperties (file: fileReadPath) }
+        def serializationClosure = { configuration -> serializeUAConfig(configuration) }
+        def inputFile = config.whitesource.configFilePath.replaceFirst('\\./', '')
+        def suffix = utils.generateSha1(config.whitesource.configFilePath)
+        def targetFile = "${inputFile}.${suffix}"
+        if(config.whitesource.productName.startsWith('DIST - ')) {
+            mapping += [
+                [name: 'checkPolicies', value: false, force: true],
+                [name: 'forceCheckAllDependencies', value: false, force: true]
+            ]
+        } else {
+            mapping += [
+                [name: 'checkPolicies', value: true, force: true],
+                [name: 'forceCheckAllDependencies', value: true, force: true]
+            ]
+        }
+        if(config.verbose)
+            mapping += [name: 'log.level', value: 'debug']
+
+        mapping += [
+            [name: 'apiKey', value: config.whitesource.orgToken, force: true],
+            [name: 'productName', value: config.whitesource.productName, force: true],
+            [name: 'productVersion', value: config.whitesource.productVersion?:'', force: true],
+            [name: 'projectName', value: config.whitesource.projectName, force: true],
+            [name: 'projectVersion', value: config.whitesource.productVersion?:'', force: true],
+            [name: 'productToken', value: config.whitesource.productToken, omitIfPresent: 'projectToken', force: true],
+            [name: 'userKey', value: config.whitesource.userKey, force: true],
+            [name: 'forceUpdate', value: true, force: true],
+            [name: 'offline', value: false, force: true],
+            [name: 'ignoreSourceFiles', value: true, force: true],
+            [name: 'resolveAllDependencies', value: false, force: true],
+            [name: 'failErrorLevel', value: 'ALL', force: true]
+        ]
+
+        // To be removed once all scenarios are fully supported
+        if(!['pip', 'golang'].contains(config.scanType))
```

Comment:
Move this to the `switch`'s default branch while removing the branches that are not yet implemented.